### ISSUE TITLE
Data persistence problem

### ISSUE THREAD

**Issue by davicedraz** [dojot/docker-compose#109]
The version v0.3.1 presents a intermittent problem related to the persistence of data.

The problem seems to be in container **dojot_persister_1** that has shown a AssertionError and does not persist the data that is received by the data broker. That is, the data comes from the sensors but is not stored.

The captured log:

```
Terminating [tenancy-watcher] ...,
Failed to bootstrap tenants's consumers:,
HTTPConnectionPool(host='auth', port=5000): Max retries exceeded with url: /admin/tenants (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f4bed7b6e50>: Failed to establish a new connection: [Errno 111] Connection refused',)),
Exiting.,
will create consumer ['kafka:9092'] history a774dc07-f4b1-4f76-b1c2-371362160bab,
Process tenancy-watcher:,
Traceback (most recent call last):,
  File "/usr/local/lib/python2.7/multiprocessing/process.py", line 267, in _bootstrap,
    self.run(),
  File "history/subscriber/kafkaSubscriber.py", line 254, in run,
    self.consumer = KafkaConsumer(bootstrap_servers=self.broker, group_id=self.group_id),
  File "/usr/local/lib/python2.7/site-packages/kafka_python-1.4.3-py2.7.egg/kafka/consumer/group.py", line 340, in __init__,
    self._client = KafkaClient(metrics=self._metrics, **self.config),
  File "/usr/local/lib/python2.7/site-packages/kafka_python-1.4.3-py2.7.egg/kafka/client_async.py", line 219, in __init__,
    self.config['api_version'] = self.check_version(timeout=check_timeout),
  File "/usr/local/lib/python2.7/site-packages/kafka_python-1.4.3-py2.7.egg/kafka/client_async.py", line 819, in check_version,
    raise Errors.NoBrokersAvailable(),
NoBrokersAvailable: NoBrokersAvailable,
Failed to bootstrap tenants's consumers:,
HTTPConnectionPool(host='auth', port=5000): Max retries exceeded with url: /admin/tenants (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fe588636e50>: Failed to establish a new connection: [Errno 111] Connection refused',)),
Exiting.,
will create consumer ['kafka:9092'] history a774dc07-f4b1-4f76-b1c2-371362160bab,
Terminating [tenancy-watcher] ...,
Failed to bootstrap tenants's consumers:,
HTTPConnectionPool(host='auth', port=5000): Max retries exceeded with url: /admin/tenants (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7ff959275e50>: Failed to establish a new connection: [Errno 111] Connection refused',)),
Exiting.,
will create consumer ['kafka:9092'] history a774dc07-f4b1-4f76-b1c2-371362160bab,
Terminating [tenancy-watcher] ...,
Failed to bootstrap tenants's consumers:,
HTTPConnectionPool(host='auth', port=5000): Max retries exceeded with url: /admin/tenants (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f2c0e992e50>: Failed to establish a new connection: [Errno 111] Connection refused',)),
Exiting.,
will create consumer ['kafka:9092'] history a774dc07-f4b1-4f76-b1c2-371362160bab,
Terminating [tenancy-watcher] ...,
Failed to bootstrap tenants's consumers:,
HTTPConnectionPool(host='auth', port=5000): Max retries exceeded with url: /admin/tenants (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fa1037dbe50>: Failed to establish a new connection: [Errno 111] Connection refused',)),
Exiting.,
will create consumer ['kafka:9092'] history a774dc07-f4b1-4f76-b1c2-371362160bab,
will create consumer ['kafka:9092'] history ae3b23fa-e8d4-4da1-a713-9e77919c8a76,
will create consumer ['kafka:9092'] history 0001d778-9b68-4a68-8526-c2baefcbf45b,
will create consumer ['kafka:9092'] history b7e91988-8ccd-4021-9058-a42059b4eeaa,
will create consumer ['kafka:9092'] history fce9239a-29e1-446d-b496-46048acf78a9,
will create consumer ['kafka:9092'] history 8c27d0a2-af68-4781-ab52-8fb5d80637e6,
will create consumer ['kafka:9092'] history a1907dcf-1696-4e8e-aa55-84156164727d,
will create consumer ['kafka:9092'] history 179d54be-18dc-4f4d-ad65-88038f3f2264,
will create consumer ['kafka:9092'] history ad9f10d8-50b7-4a91-bd57-62c7c0c9d983,
kafka consumer created 0001d778-9b68-4a68-8526-c2baefcbf45b - 4.9988090992,
kafka consumer created ae3b23fa-e8d4-4da1-a713-9e77919c8a76 - 8.9327249527,
kafka consumer created 8c27d0a2-af68-4781-ab52-8fb5d80637e6 - 8.85845112801,
kafka consumer created b7e91988-8ccd-4021-9058-a42059b4eeaa - 8.90321087837,
kafka consumer created 179d54be-18dc-4f4d-ad65-88038f3f2264 - 8.81261301041,
kafka consumer created fce9239a-29e1-446d-b496-46048acf78a9 - 8.891092062,
kafka consumer created a1907dcf-1696-4e8e-aa55-84156164727d - 8.84525609016,
kafka consumer created a774dc07-f4b1-4f76-b1c2-371362160bab - 8.96901202202,
kafka consumer created ad9f10d8-50b7-4a91-bd57-62c7c0c9d983 - 8.80308103561,
Process device-data-watcher:,
Traceback (most recent call last):,
  File "/usr/local/lib/python2.7/multiprocessing/process.py", line 267, in _bootstrap,
    self.run(),
  File "history/subscriber/kafkaSubscriber.py", line 258, in run,
    for message in self.consumer:,
  File "/usr/local/lib/python2.7/site-packages/kafka_python-1.4.3-py2.7.egg/kafka/vendor/six.py", line 561, in next,
    return type(self).__next__(self),
  File "/usr/local/lib/python2.7/site-packages/kafka_python-1.4.3-py2.7.egg/kafka/consumer/group.py", line 1118, in __next__,
    return next(self._iterator),
  File "/usr/local/lib/python2.7/site-packages/kafka_python-1.4.3-py2.7.egg/kafka/consumer/group.py", line 1052, in _message_generator,
    self._coordinator.poll(),
  File "/usr/local/lib/python2.7/site-packages/kafka_python-1.4.3-py2.7.egg/kafka/coordinator/consumer.py", line 281, in poll,
    self.ensure_active_group(),
  File "/usr/local/lib/python2.7/site-packages/kafka_python-1.4.3-py2.7.egg/kafka/coordinator/base.py", line 386, in ensure_active_group,
    future = self._send_join_group_request(),
  File "/usr/local/lib/python2.7/site-packages/kafka_python-1.4.3-py2.7.egg/kafka/coordinator/base.py", line 468, in _send_join_group_request,
    _f = self._client.send(self.coordinator_id, request),
  File "/usr/local/lib/python2.7/site-packages/kafka_python-1.4.3-py2.7.egg/kafka/client_async.py", line 504, in send,
    if not self._maybe_connect(node_id):,
  File "/usr/local/lib/python2.7/site-packages/kafka_python-1.4.3-py2.7.egg/kafka/client_async.py", line 340, in _maybe_connect,
    assert broker, 'Broker id %s not in current metadata' % node_id,
AssertionError: Broker id None not in current metadata,
Process device-data-watcher:,
Traceback (most recent call last):,
  File "/usr/local/lib/python2.7/multiprocessing/process.py", line 267, in _bootstrap,
    self.run(),
  File "history/subscriber/kafkaSubscriber.py", line 258, in run,
    for message in self.consumer:,
  File "/usr/local/lib/python2.7/site-packages/kafka_python-1.4.3-py2.7.egg/kafka/vendor/six.py", line 561, in next,
    return type(self).__next__(self),
  File "/usr/local/lib/python2.7/site-packages/kafka_python-1.4.3-py2.7.egg/kafka/consumer/group.py", line 1118, in __next__,
    return next(self._iterator),
  File "/usr/local/lib/python2.7/site-packages/kafka_python-1.4.3-py2.7.egg/kafka/consumer/group.py", line 1052, in _message_generator,
    self._coordinator.poll(),
  File "/usr/local/lib/python2.7/site-packages/kafka_python-1.4.3-py2.7.egg/kafka/coordinator/consumer.py", line 281, in poll,
    self.ensure_active_group(),
  File "/usr/local/lib/python2.7/site-packages/kafka_python-1.4.3-py2.7.egg/kafka/coordinator/base.py", line 386, in ensure_active_group,
    future = self._send_join_group_request(),
  File "/usr/local/lib/python2.7/site-packages/kafka_python-1.4.3-py2.7.egg/kafka/coordinator/base.py", line 468, in _send_join_group_request,
    _f = self._client.send(self.coordinator_id, request),
  File "/usr/local/lib/python2.7/site-packages/kafka_python-1.4.3-py2.7.egg/kafka/client_async.py", line 504, in send,
    if not self._maybe_connect(node_id):,
  File "/usr/local/lib/python2.7/site-packages/kafka_python-1.4.3-py2.7.egg/kafka/client_async.py", line 340, in _maybe_connect,
    assert broker, 'Broker id %s not in current metadata' % node_id,
AssertionError: Broker id None not in current metadata,
```

**Comment by victorpraxedes** [comment]
https://github.com/dojot/docker-compose/issues/109

This issue might be related

**Comment by danielnunesdeoliveira** [comment]
I am similar problem, but I am using v0.30 version.

1) Log compose
![image](https://user-images.githubusercontent.com/29581014/69370207-41f0ce80-0c7c-11ea-9f7c-54ebca72a11c.png)
2) Log Persister
![image](https://user-images.githubusercontent.com/29581014/69370241-53d27180-0c7c-11ea-8dc9-ea51f9927c57.png)

Can someone help me?

**Comment by danielnunesdeoliveira** [comment]
My problem was resolved. The _persister_ container was changed by me.

**Comment by davicedraz** [comment]
Closing this #109 since was fixed in version v0.4.2
